### DISCUSSION TITLE
Health check HTTP request

### DISCUSSION THREAD

**Discussion by Walker555** [ArcadeData/arcadedb#661] [Q&A] [answered]
We use OrientDB 3.x and want to migrate to ArcadeDB. The OrientDB allows an unauthenticated HTTP request  `server_ip:2480/server/version` that we use as public health check of the server. Is something similar available in Arcade?

**Comment by lvca** [comment, marked as answer]
You can call the `/server/` HTTP API:

curl -X GET http://localhost:2480/api/v1/server --user root:arcadedb-password

Look at: https://docs.arcadedb.com/#HTTP-ServerInfo

Unfortunately, it requires a server user. We believe it's risky to expose server information to the public for attacks.

Apart from the root user created the first time you run the server, how do you create a new user? Unfortunately, we didn't provide a script to manage users easily. We should do that from the console. I'm going to open an issue about it.

Via code:

```java
ArcadeDBServer server = new ArcadeDBServer();
server.start();
...
server.getSecurity().createUser(new JSONObject().put("name", "elon").put("password", SECURITY.encodePassword("musk")));
```

**Reply by lvca** [reply]
Found this issue already created by @gramian: https://github.com/ArcadeData/arcadedb/issues/574.

**Reply by gramian** [reply]
Would an endpoint that just responds a status `200` header to a `GET` request also be a security problem?
[Benthos](https://www.benthos.dev/docs/components/http/about#endpoints) has such a `/ready` endpoint.

**Reply by Walker555** [reply]
> Would an endpoint that just responds a status `200` header to a `GET` request also be a security problem? [Benthos](https://www.benthos.dev/docs/components/http/about#endpoints) has such a `/ready` endpoint.

Exactly, I would appreciate if ArcadeDB could implement a `/ready` endpoint.

**Reply by lvca** [reply]
Ok, it makes sense. Created a new issue for that: https://github.com/ArcadeData/arcadedb/issues/665

**Reply by lvca** [reply]
Ok, implemented and tested. It's in the main branch ready to be released with 22.12.1 in about a week or so. You can now call `/ready` without providing any authentication. The return has error code 204 (OK, no content) with no content.
